app/controllers/api/custom_buttons_controller.rb

```diff
@@ -44,14 +43,11 @@ def build_custom_button_options
 
     def create_or_update_resource_action(data)
       return nil if data.empty?
-      id = data["id"]
+      id = data.delete("id")
       if id.present?
-        resource_action = ResourceAction.find(id)
-        data.except!("id")
-        resource_action.update!(data.deep_symbolize_keys)
-        resource_action
+        ResourceAction.find(id).tap { |ra| ra.attributes = data }
```

Comment:
Still feels awkward to me that we're relying on `has_one :resource_action, :autosave => true` to save the resource action when `cb.update!` is called.